File: Codeforces/div806/E.cpp

#include <iostream>
#include <vector>
#include <string>

using namespace::std;

void rotate(int &x, int &y, int n)
{
	int xx = x;
	x = n - y - 1;
	y = xx;
}

int main()
{
	int t;
	cin >> t;
	while (t-- > 0)
	{
		int n;
		cin >> n;
		vector<string> m(n);
		for (int y=0; y < n; ++y)
		{
			cin >> m[y];
			for (auto &c : m[y])
				c -= '0';
		}
		int ans = 0;
		for (int x=0; x < n; ++x)
		{
			for (int y=0; y < n; ++y)
			{
				int z = 0;
				z += m[y][x]; // 1
				int xx = x;
				int yy = y;
				rotate(xx, yy, n);
				z += m[yy][xx]; // 2
				rotate(xx, yy, n);
				z += m[yy][xx]; // 3
				rotate(xx, yy, n);
				z += m[yy][xx]; // 4
				ans += std::min(z, 4 - z);
			}
		}
		cout << ans / 4 << endl;
	}
}

File: Codeforces/div806/E.py

def rotate(x, y, n):
    return n - y - 1, x


for _ in range(int(input())):
    n = int(input())
    m = [list(map(int, input())) for _ in range(n)]
    ans = 0
    for _y in range(n):
        for _x in range(n):
            x, y = _x, _y
            r = list()
            r.append(m[y][x])
            x, y = rotate(x, y, n)
            r.append(m[y][x])
            x, y = rotate(x, y, n)
            r.append(m[y][x])
            x, y = rotate(x, y, n)
            r.append(m[y][x])
            z = r.count(0)
            # print(min(z, 4 - z))
            ans += min(z, 4 - z)
            # if z > 2:
            #     ans += 4 - z
            # else:
            #     ans += z
    print(ans // 4)
